### PR TITLE
YOLOv5 ISICs Problem

### PR DESCRIPTION
**Pull request by RTongs**
This is a solution of identifying melanoma in the ISICs 2018 dataset using YOLOv5. I started out with Mask R-CNN but found it much too time consuming to try to convert it to TensorFlow 2. Maybe YOLOv3 or v1 would have been more appropriate, but I found this out too late. Most of my time was spent trying to do data augmentation, generating the masks, data splitting, and understanding how to use YOLOv5.

The technicalities of the data augmentation included rotations and flips. With the data splitting just being a standard 80/20 on training and test, and then validation taking 20% of the 80% of the training set.

I realise the importance of evaluating with a test set, but I was only able to test my only working model on just the validation set as I lost my model checkpoints and ran out of time. I did get the test inferencing working though, you are welcome to try it.

I was not able to fully finish my assignment due to unforeseen medical issues so apologies for the incomplete assignment. Even finishing the report was physically painful and damaging. Note that my username is ryemigie for one of my GitHub accounts which is why it came up. Apologies.

### CODE REVIEW

**Comment by shakes76** [comment]
## TF/Torch Usage
Adequate use and implementation
Late submission, but medical reasons
## Good Practice (Design/Commenting)
Good spacing and comments
No Header blocks -1
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage 
Module present, but not applied to test data -1
No Data leakage 
Difficulty: Normal -5
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
Good Description and Comments 
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
Please remove WandB files, these will clog up the repository. Your pull request will not be merged until they are removed. Feedback marks will be awarded when these are removed. -2
Request Description could be more informative (technical aspects i.e.) -1

**Comment by RTongs** [comment]
Just updating the PR now.

**Comment by RTongs** [comment]
The merge conflicts is some weird bug I've been getting... it keeps mentioning recognition/ISICs_Unet/README.md which I have not touched at all. I'll try and fix now.

**Comment by RTongs** [comment]
Just made a minor update to my initial comment, and now I'm ready to merge. I deleted the wandb files too.

**Comment by shakes76** [comment]
Feedback marks awarded.

**Comment by shakes76** [comment]
Still can't merge though because of the .cache files present that take up some space.

**Comment by RTongs** [comment]
Deleted cache files, ready to merge.
